src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+            currentNode,
+            options.connectedHost,
+            options.connectedPort,
+            relayPublicKey,
+          );
+        }
+
+        if (successful) {
+          pkLogger('peer info was successfully updated', PKMessageType.SUCCESS);
+        } else {
+          pkLogger('something went wrong, peer info could not be updated', PKMessageType.WARNING);
+        }
+      }),
+    );
+}
+function makeGetPeerInfoCommand() {
+  return new commander.Command('get')
+    .description('get the peer info for a particular public key')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-b, --base64', 'output peer info as a base64 string')
+    .option('-c, --current-node', 'only list the peer information for the current node, useful for sharing')
+    .option('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const currentNode: boolean = options.currentNode;
+        const publicKeyPath = options.publicKey;
+        const base64 = options.base64;
+        // read in publicKey if it exists
+        let publicKey: string | undefined;
+        if (publicKeyPath) {
+          publicKey = fs.readFileSync(publicKeyPath).toString();
+        }
+
+        const peerInfo: PeerInfo = await client.getPeerInfo(nodePath, currentNode, publicKey);
+
+        if (base64) {
+          pkLogger(peerInfo.toStringB64(), PKMessageType.SUCCESS);
+        } else {
+          pkLogger('Peer Public Key:', PKMessageType.INFO);
+          pkLogger(peerInfo.publicKey, PKMessageType.SUCCESS);
+
+          pkLogger('Peer Connected Address:', PKMessageType.INFO);
+          pkLogger(peerInfo.connectedAddress?.toString() ?? '', PKMessageType.SUCCESS);
+
+          pkLogger('Relay Public Key:', PKMessageType.INFO);
+          pkLogger(peerInfo.relayPublicKey ?? '', PKMessageType.SUCCESS);
+        }
+      }),
+    );
+}
+
+function makeListPeersCommand() {
+  return new commander.Command('list')
+    .description('list all connected peers')
+    .alias('ls')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+
+        const publicKeys = await client.listPeers(nodePath);
+        if (publicKeys === undefined || publicKeys.length == 0) {
```

Comment:
I suggest using `null` as your sentinel value to represent `Maybe` as in Haskell. Use `undefined` only in the case where non-existence is actually what is expected. That is I only use `undefined` for side-effectful functions that do not have return values (this is ideal). And also for checking for key existence. In all other cases `null`.

However for lists, the sentinel value should be `[]`, not `null`. This keeps the types simpler. And also means you do not need to check for sentinel values. You can just compose with iteration and if you get nothing, you get nothing...